backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/codec/impl/ElasticsearchYearMonthFieldCodec.java

```diff
@@ -18,4 +20,9 @@ public ElasticsearchYearMonthFieldCodec(DateTimeFormatter delegate) {
 	protected YearMonth nullUnsafeParse(String stringValue) {
 		return YearMonth.parse( stringValue, formatter );
 	}
+
+	@Override
+	protected Long nullUnsafeScalar(YearMonth value) {
+		return value.atDay( 1 ).atStartOfDay().atZone( ZoneOffset.UTC ).toInstant().toEpochMilli();
```

Comment:
You can just call `value.toInstant( ZoneOffset.UTC )` instead of `atZone(...).toInstant()`.